Fixed import errors and updated to support python 3

angeryrohan opened the pull request:
Fixes #121 

- Imported queue as queue to avoid import error

![import](https://user-images.githubusercontent.com/29266591/47969668-19158f80-e0a1-11e8-8ebd-4292538e4bc2.png)


- Enabled backward compatible flask server. Did this by adding parenthesis around every print command in dronepool.py

angeryrohan commented:
Earlier:

![ree](https://user-images.githubusercontent.com/29266591/49377191-7ac42a80-f72f-11e8-8b59-c15f18e9c6f8.png)

Now:

![ree 2](https://user-images.githubusercontent.com/29266591/49377210-87e11980-f72f-11e8-988c-2e8dc955f20e.png)

rohancl commented:
@OshanIvantha   What do you think of the changes ?

angeryrohan commented:
## End to End test for feature - [LogIn Authorization]<br> + Login Vulnerability Test
In the recent end to end testing of DroneSym, I wrote tests for ensuring URL protection by redirections and making sure that the login system works with both, user's and admin's credentials. Logged in and out several times, in different ways to ensure  security, the functioning of Login's UI and proper authorization.

## A look


![ezgif com-gif-maker 1](https://user-images.githubusercontent.com/29266591/49705213-0a436f00-fc43-11e8-92d5-619e9285270c.gif)

angeryrohan commented:
@OshanIvantha Please Review,
Thanks ^.^

swatisingh45 reviewed (commented):
Squash your commits, please.

angeryrohan commented:
@swatisingh45 Sure, I'm on it...

sudodaksh committed:
Fixed import errors and updated to support python 3

- Imported queue as queue to avoid import error

- Enabled backward compatible flask server

Updated main.py
Removed useless import
Made the npm build more clean

Removed depricated use of open() + made port flexibleRemoved the warning to get a clean build:  DeprecationWarning: `open()` is deprecated in mongoose >= 4.11.0 by declaring the boolean value of Mongoclient as true.

What I personally encountered... I was running another server on port 3000, Thus when I was setting up Dronesym, I got an error of port already being in use, So I made a global variable named port, and added string concatenation to console the running port into the terminal. If users already have port 3000 in use, then they could just change the value of variable in /dronesym-node/index.js.
Added missing instruction of running mongod

Most of the people including me forgot to run ```mongod``` in one terminal, and then execute ```mongorestore --db dronesym dronedb/dronesym```, So I've added an instruction step.
Added latest and more supportive version of flask
Added E2E tests for Login authorization 

1. Checks URL redirection vulnerability (security purpose).
2. Checks Server's availability.
3. Checks Login authorization through admin's account.
4. Checks the dashboard appearance on putting correct credentials.
5. Logs out of admin's account 
6. Checks Login authorization through admin's account.

angeryrohan commented:
@swatisingh45 Done.

angeryrohan committed:
Added character encoding & meta for responsiveness

angeryrohan committed:
Made the Login Page responsive

This DroneSym Webapp can now work on small phones, mid-sized phones, large phones, laptops, large laptops, HD TV and even a 4K TV

angeryrohan commented:
Recently, I made the page responsive at every scale, 
The web app now works flawlessly irrespective of the screen's dimensions.
EARLIER: 

![screenshot from 2018-12-15 03-57-08](https://user-images.githubusercontent.com/29266591/50033230-8fe96500-001d-11e9-9b9f-367508f94f74.png)

NOW:

![webp net-gifmaker](https://user-images.githubusercontent.com/29266591/50033177-6a5c5b80-001d-11e9-880f-1a6c1eb8c279.gif)

angeryrohan commented:
@OshanIvantha Please Review ^.^

agentmilindu commented on dronesym-frontend/src/app/login/login.component.html:
```diff
@@ -1,15 +1,17 @@
 			<div class="card-image">
-				<div class="logo-container">
-					<img src="./assets/img/dronesym_logo.png"/>
+				<div class="logo-container"><!-- -->
```
Do we need these `<!-- -->`?

agentmilindu commented on dronesym-node/index.js:
```diff
@@ -57,7 +59,7 @@ app.use(function(err, req, res, next) {
 
 
 http.listen(3000, function(){
```
I think you have to use your new `port` variable here also if you really want it to be used

angeryrohan replied:
No we don't need these, This was just to test something.... I'll remove them right away ^.^

agentmilindu commented:
I see lot of files related to `WiredTiger`, do we need them?

angeryrohan commented:
Sorry, that's a terrible mistake there! I'll fix it right now....

angeryrohan committed:
Removed redundant comments

angeryrohan committed:
Update index.js

angeryrohan commented:
@agentmilindu , I have made the required changes..... 
And where exactly did you see the ```WiredTiger``` files, Please give me the location

agentmilindu commented:
@rohancl  See the higlighted places. Plus, I assuemm `.wt` files are also related to WildTiger.
 
![image](https://user-images.githubusercontent.com/1391468/50053212-f45c0f80-0156-11e9-8d19-3024030f9048.png)

angeryrohan commented:
@agentmilindu , Seems that that was an auto-generated file by MongoDB, Anyway I've deleted them... Have a look now ^.^

agentmilindu reviewed (approved):
LGTM

agentmilindu commented:
@OshanIvantha Can you check and merge this?

angeryrohan commented:
@agentmilindu @OshanIvantha @malithsen  All required code quality tests are now passing ^.^ ....

ivantha reviewed (changes requested):
@rohancl This looks good to go. 
However can you squash the commits please? 
Also there is an unnecessary package-lock.json file created at the root.

angeryrohan committed:
Deleted Unnecessary Cache files and improved code quality

Delete WiredTiger.turtle
Delete WiredTiger.lock
Delete WiredTiger.wt
Delete WiredTigerLAS.wt
Delete _mdb_catalog.wt
Delete collection-0--2321288784840600282.wt
Delete storage.bson
Deleted unecessary WiredTiger stuff
deleted Redundant stuff
Delete collection-2-7301280593392019203.wt
Delete collection-4--2321288784840600282.wt
Delete metrics.2018-11-04T20-24-09Z-00000
Delete index-1--2321288784840600282.wt
Delete sizeStorer.wt
Delete mongod.lock
Delete index-1-7301280593392019203.wt
Delete index-3--2321288784840600282.wt
Delete WiredTigerLog.0000000002
Delete WiredTigerPreplog.0000000001
Delete WiredTigerPreplog.0000000002
Delete metrics.2018-11-04T20-24-37Z-00000
Delete metrics.interim
Delete index-3-7301280593392019203.wt
Delete index-4-7301280593392019203.wt
Delete index-5--2321288784840600282.wt
Disabled no console warning for Codacy's nodeJS code compiling.
Replaced single quotes with double quotes for Codacy's Code quality
Update package-lock.json
Removed package-lock.JSON

angeryrohan commented:
Oh, thanks for pointing that out...
I removed the package-lock.JSON file and squashed the commits
Thanks, ^.^
@OshanIvantha
